services/app-api/src/postgres/contractAndRates/prismaSharedContractRateHelpers.ts

```diff
@@ -198,19 +198,35 @@ const getParentContractID = (
         // that can be replaced in higher places.
         parentContractID = 'DRAFT_PARENT_REPLACE_ME'
     } else {
-        // check the initial submission
-        if (firstRevision.relatedSubmissions.length == 0) {
-            console.info('No related submission. Unmigrated rate.')
-            parentContractID = '00000000-1111-2222-3333-444444444444'
-        } else {
-            if (submission.submittedContracts.length !== 1) {
-                const msg =
-                    'programming error: its a submitted rate that was not submitted with a contract initially'
-                console.error(msg)
-                return new Error(msg)
+        //find the latest submitted revision with submitted contracts
+        const latestSubmittedRevision = revisions.find(
+            (revision) => revision.submitInfo?.submittedContracts[0]
+        )
+
+        if (latestSubmittedRevision?.submitInfo) {
+            if (latestSubmittedRevision.relatedSubmissions.length == 0) {
+                console.info('No related submission. Unmigrated rate.')
+                parentContractID = '00000000-1111-2222-3333-444444444444'
+            } else {
+                // there should always only be one submitted contract at the revision level submit info
+                const latestParentContract =
+                    latestSubmittedRevision.submitInfo.submittedContracts[0]
+
+                if (
+                    latestSubmittedRevision.submitInfo?.submittedContracts
+                        .length !== 1
+                ) {
+                    return new Error(
+                        'programming error: its a submitted rate that was not submitted with a contract initially'
+                    )
+                }
+
+                parentContractID = latestParentContract.contractID
             }
-            const firstContract = submission.submittedContracts[0]
-            parentContractID = firstContract.contractID
+        } else {
+            return new Error(
+                'Could not find a submitted revision to get parent contract id.'
```

Comment:
nit: Return this error at the top in a guard clause. Basically the whole point of the `if ...else` is to guard against nonsensical data because we have overly loose database types -- is that right?

 If so, do the data checks at the top and return promptly. Want to avoid the doubly nested if ...else, it's confusing.